tests/test_parser.py

```diff
@@ -0,0 +1,58 @@
+import types
+
+import pytest
+
+from nexus.parser import Nexus, TokenType
+
+
+@pytest.mark.parametrize(
+    'nex,expect,roundtrip',
+    [
+        (
+            """#NEXUS
+BEGIN TREES;
+TREE best = (fish , (frog,
+(snake, mouse))) ;
+END;""",
+            lambda n: sum(1 for t in n.nexus if t.type != TokenType.WHITESPACE) == 23,
+            True),
```

Comment:
This is an example taken from the Nexus paper, together with the expected token count of 23.

Reply:
great!